### PR TITLE
Persist microsoft user profile in users table

### LINKED ISSUE #23: Persist microsoft user profile in users table
- [x] introduce ENV variable "ADMIN_USERS" as comma separated list of uuids
- [x] save profile/user object into variable
- [x] check if user exists in database
- [x] if not found: persist user object to database
- [x]  if found: upsert user entry

### PR DESCRIPTION
**Pull request by curiosdevcookie**
- Set host to env variable.
- Add debug route for microsoft user.
- Destructure the microsoft profile object according to the needed values.
- Save authenticated user into variable.
- Add env variable ADMIN_USERS.
- Persist user object to database.

Closes #23.

### CODE REVIEW

**Review by phaus — COMMENTED** [review]
Funktioniert alles. Sehr cool!
Meine Anmerkungen sind alles nur Detail-Wünsche 🙈 
Du kannst ja mal gucken, ob Dir davon etwas gefällt :-)

File: db/migrations/20230718063720_users_table.js
[the remark this answers is not in the document]
Comment (curiosdevcookie) [reply db/migrations/20230718063720_users_table.js:1]: I don't modify the original table? The column is_admin was there from the start.

File: src/insertOrUpdateUser.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  connection: {
+    host: process.env.DB_HOST,
+    port: Number(process.env.POSTGRES_PORT),
+    user: process.env.POSTGRES_USER,
+    password: process.env.POSTGRES_PASSWORD,
+    database: process.env.POSTGRES_DB,
+  },
+});
+
+export async function insertOrUpdateUser(req: Request) {
```
[the remark this answers is not in the document]
Comment (curiosdevcookie) [reply src/insertOrUpdateUser.ts:17]: You have to show me how to make handing in the user work, please.

Actually, unfortunately, the struct is already in the config `knexfile.js`, but I wasn't able to load it from there, so I had to stick it inside there as well :-(

File: src/insertOrUpdateUser.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const adminUserIds = process.env.ADMIN_USERS?.split(",") || [];
+    const isAdmin = adminUserIds.includes(authenticatedUser.userId);
+
+    authenticatedUser.is_admin = isAdmin;
+
+    try {
+      await database("users")
+        .insert({
+          user_id: authenticatedUser.userId,
+          provider: "Microsoft",
```
[the remark this answers is not in the document]
Comment (curiosdevcookie) [reply src/insertOrUpdateUser.ts:29]: How?

File: src/insertOrUpdateUser.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (curiosdevcookie) [reply src/insertOrUpdateUser.ts:17]: Why check for null?

File: src/models/user.ts [hunk 3]
```diff
@@ -10,14 +15,23 @@ export class User {
...
+    is_admin: boolean
+  ) {
     this.userId = userId;
     this.displayName = displayName;
     this.mail = mail;
-    this.userPrincipalName = userPrincipalName;
+    this.givenName = givenName;
+    this.surname = surname;
+    this.is_admin = is_admin;
   }
```
[the remark this answers is not in the document]
Comment (curiosdevcookie) [reply src/models/user.ts:36]: I don't see the advantage in this mixed approach?

File: src/insertOrUpdateUser.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (curiosdevcookie) [reply src/insertOrUpdateUser.ts:29]: Done!

**Pull request merged**